sql/resource_query.go

```diff
@@ -0,0 +1,169 @@
+	m.SchemaPath("warehouse_id").SetRequired().SetValidateFunc(validation.StringIsNotWhiteSpace)
+	m.SchemaPath("parent_path").SetCustomSuppressDiff(common.WorkspaceOrEmptyPathPrefixDiffSuppress).SetForceNew()
+	m.SchemaPath("owner_user_name").SetSuppressDiff()
+	m.SchemaPath("run_as_mode").SetSuppressDiff()
+	//m.SchemaPath("").SetSuppressDiff()
+	//m.SchemaPath("").SetSuppressDiff()
+	m.SchemaPath("id").SetReadOnly()
+	m.SchemaPath("create_time").SetReadOnly()
+	m.SchemaPath("lifecycle_state").SetReadOnly()
+	m.SchemaPath("last_modifier_user_name").SetReadOnly()
+	m.SchemaPath("update_time").SetReadOnly()
+
+	// customize parameters
+	m.SchemaPath("parameter", "name").SetRequired().SetValidateFunc(validation.StringIsNotWhiteSpace)
+	m.SchemaPath("parameter", "date_range_value", "precision").SetSuppressDiff()
+	m.SchemaPath("parameter", "date_value", "precision").SetSuppressDiff()
+	m.SchemaPath("parameter", "query_backed_value", "query_id").SetRequired()
+	m.SchemaPath("parameter", "text_value", "value").SetRequired()
+	m.SchemaPath("parameter", "numeric_value", "value").SetRequired()
+	// TODO: fix setting of AtLeastOneOf
+	// valuesAlof := []string{
+	// 	"parameter.0.date_range_value",
+	// 	"parameter.0.date_value",
+	// 	"parameter.0.query_backed_value",
+	// 	"parameter.0.text_value",
+	// 	"parameter.0.numeric_value",
+	// 	"parameter.0.enum_value",
+	// }
+	// for _, f := range valuesAlof {
+	// 	m.SchemaPath("parameter", strings.TrimPrefix(f, "parameter.0.")).SetAtLeastOneOf(valuesAlof)
+	// }
+	return m
+}
+
+type queryCreateStruct struct {
+	sql.CreateQueryRequestQuery
+}
+
+func (queryCreateStruct) Aliases() map[string]map[string]string {
+	return map[string]map[string]string{
+		"sql.queryCreateStruct": queryAliasMap,
+	}
+}
+
+func (queryCreateStruct) CustomizeSchema(s *common.CustomizableSchema) *common.CustomizableSchema {
+	return s
+}
+
+type queryUpdateStruct struct {
+	sql.UpdateQueryRequestQuery
+}
+
+func (queryUpdateStruct) Aliases() map[string]map[string]string {
+	return map[string]map[string]string{
+		"sql.queryUpdateStruct": queryAliasMap,
+	}
+}
+
+func (queryUpdateStruct) CustomizeSchema(s *common.CustomizableSchema) *common.CustomizableSchema {
+	return s
+}
+
+func ResourceQuery() common.Resource {
+	s := common.StructToSchema(queryStruct{}, nil)
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var q queryCreateStruct
+			common.DataToStructPointer(d, s, &q)
+			apiQuery, err := w.Queries.Create(ctx, sql.CreateQueryRequest{
+				Query: &q.CreateQueryRequestQuery,
+			})
+			if err != nil {
+				return err
+			}
+			d.SetId(apiQuery.Id)
+			owner := d.Get("owner_user_name").(string)
+			if owner != "" {
+				_, err = w.Queries.Update(ctx, sql.UpdateQueryRequest{
+					Query: &sql.UpdateQueryRequestQuery{
+						OwnerUserName: owner,
```

Comment:
is this another set of APIs where the owner cannot be set on create?

Reply:
yep - same as for alerts